.github/workflows/My-ci-pipeline.yml

```diff
@@ -28,4 +28,79 @@ jobs:
       run: docker build --target test --tag todo-app:test .
     
     - name: BuildAndTestImage
-      run: docker run --env-file .env.test todo-app:test tests
\ No newline at end of file
+      run: docker run --env-file .env.test todo-app:test tests
+
+  Publish:
+    if: ${{ github.ref == 'refs/heads/main' && github.event_name == 'push' }}
+    #if: ${{ github.ref == 'refs/heads/module8' && github.event_name == 'push' }}
+    name: Publish
+    runs-on: ubuntu-latest
+    needs: [build]
+    steps:
+    - uses: actions/checkout@v2
+    
+    - uses: satackey/action-docker-layer-caching@v0.0.11
+      continue-on-error: true
+
+    - name: publish to docker hub
+      run: echo "Publishing!"
+
+    #- name: Build Image
+    #  run: docker build --tag latest .
+    - name: Build Image
+      run: docker build --target production --tag moudule8 .
+
+    - name: Change Tag
+      run: docker tag moudule8 wishy78/module8:module8
+
+    - name: Login to Docker hub
+      run: docker login --username wishy78 --password "${{ secrets.DOCKER_PASSWORD }}"
```

Comment:
We should ideally action this step using the `--password-stdin` flag rather than passing the password explicitly, to avoid leaking those credentials to shell logs. [See the docs](https://docs.docker.com/engine/reference/commandline/login/) for more info, example of how that might look below:
```
echo "${{ secrets.DOCKER_PASSWORD }}" | docker login --username wishy78 --password-stdin
```